examples/cpp/pyperf/PyPerfNativeStackTrace.cc

```diff
@@ -65,23 +81,33 @@ NativeStackTrace::NativeStackTrace(uint32_t pid, const unsigned char *raw_stack,
     goto out;
   }
 
+  struct bcc_symbol resolved_symbol;
+  if (NativeStackTrace::insert_dso_name) {
+    procSymbols = get_proc_symbols(pid);
+  }
+
   do {
     unw_word_t offset;
-    char sym[256];
-    char   *realname;
-    int     status;
-
     // TODO: This function is very heavy. We should try to do some caching here, maybe in the
     //       underlying UPT function.
     res = unw_get_proc_name(&cursor, sym, sizeof(sym), &offset);
```

Comment:
A slight optimization - this can move to the non-DSO branch, because if we use `ProcSyms,` then this result isn't used at all, right? Technically we can unwind with libunwind and symbolicate with BCC.

Reply:
Yes, I believe that should work this way.

Reply:
Adopted the suggested optimization.

Reply:
Cool